nsv13/code/controllers/subsystem/starsystem.dm

```diff
@@ -86,17 +91,130 @@ Returns a faction datum by its name (case insensitive!)
+				visitable = LAZYACCESS(sys_info, "visitable") || TRUE,
+				is_hypergate = LAZYACCESS(sys_info,"is_hypergate") || FALSE,
+				preset_trader = (LAZYACCESS(sys_info,"preset_trader")) ? text2path(sys_info["preset_trader"]) : null,
+				system_traits = LAZYACCESS(sys_info,"system_traits") || list(),
+				system_type = (LAZYACCESS(sys_info,"system_type") && sys_info["system_type"] != "null" && sys_info["system_type"] != null) ? json_decode(sys_info["system_type"]) : list(),
+				audio_cues = (LAZYACCESS(sys_info,"audio_cues") && sys_info["audio_cues"] != "null" && sys_info["audio_cues"] != null) ? json_decode(sys_info["audio_cues"]) : list(),
+				wormhole_connections = (LAZYACCESS(sys_info,"wormhole_connections") && sys_info["wormhole_connections"] != "null" && sys_info["wormhole_connections"] != null) ? json_decode(sys_info["wormhole_connections"]) : list(),
+				startup_proc = LAZYACCESS(sys_info,"startup_proc") || null
+				//Future implementation ideas: We can cache system_contents. I'm not dealing with that now though.
+			)
+			systems += next
+		}
+		catch(var/exception/e){
+			message_admins("WARNING: Invalid star system in json: [sys_info["name"]] ([e]). Skipping...")
+			continue
+		}
+	message_admins("Successfully loaded starmap layout from [_source_path]")
+
+
+/datum/controller/subsystem/star_system/proc/instantiate_systems_backup()
 	for(var/instance in subtypesof(/datum/star_system))
 		var/datum/star_system/S = new instance
 		if(S.name)
 			systems += S
+	if(saving)
+		return
+	saving = TRUE
+	save()
+	saving = FALSE
+/**
+<summary>Save the current starmap layout to a json file. Used for persistence.</summary>
+<param></param>
+*/
+
+/datum/controller/subsystem/star_system/proc/save(_destination_path = "config/starmap/starmap_default.json")
+	message_admins("Saving current starsystem layout...")
+	var/json_file = null
+	//Read the file in...
+	try
+		json_file = file(_destination_path)
+	catch(var/exception/ex)
+		message_admins("WARNING: Unable to open [_destination_path]: [ex]")
+		return 1
+	var/list/file_data = list()
+	for(var/datum/star_system/S in systems)
+		if(S == null || istype(S, /datum/star_system/random))
+			continue
+		var/list/entry = list(
+			//Fluff.
+			"name"=S.name,
+			"desc"=S.desc,
+			"threat_level"=S.threat_level,
+			//General system props
+			"alignment" = S.alignment,
+			"hidden"=S.hidden,
+			"system_type" = json_encode(S.system_type),
+			"system_traits"=S.system_traits,
+			"is_capital"=S.is_capital,
+			"adjacency_list"=json_encode(S.adjacency_list),
+			"wormhole_connections"=json_encode(S.wormhole_connections),
+			"fleet_type" = S.fleet_type,
+			//Coords, props.
+			"x" = S.x,
+			"y" = S.y,
+			"parallax_property"=S.parallax_property,
+			"visitable"=S.visitable,
+			"sector"=S.sector,
+			"is_hypergate"=S.is_hypergate,
+			"preset_trader"=S.preset_trader,
+			"audio_cues" = json_encode(S.audio_cues),
+			"startup_proc" = S.startup_proc
+		)
+		file_data[++file_data.len] = entry
```

Comment:
```suggestion
		file_data += entry
```

Reply:
Weirdly, I found that didn't work.

I picked this trick up off monster, we have a load of it on our tgui backends. Is this incorrect?

Reply:
Nope, turns out you're totally correct!
http://www.byond.com/forum/post/1955485

Now that I recall, I tried to change one of the fighter(?) tgui data procs and broke it by doing this lmao